### ISSUE TITLE
regression: editor hangs for ~0.5sec after inserting a space

### ISSUE THREAD

**Issue by benswift** [zed-industries/zed#14851]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

About 1 week ago (sorry, I didn't pay attention to the exact day/release it started happening) zed started freezing/hanging for around half a second whenever I entered a space (using the spacebar). This was not happening previously (and it's so noticeable because everything else in Zed is super fast).

The freeze happens with co-pilot signed in and signed out, and happens both for editing files and also in the Terminal. It also happens across all different languages/LSPs.

Happy to provide/gather more info if it's helpful in tracking down the issue.

### Environment

Zed: v0.144.4 (Zed)
OS: macOS 14.5.0
Memory: 128 GiB
Architecture: aarch64

### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary><pre>

</pre></details>

**Comment by osiewicz** [comment]
Could you provide a repro case?

**Comment by benswift** [comment]
Hey there

In putting together the repro case, I found out it was caused by a change in my config. So it's not a hang after all, and sorry for the hassle.

For the benefit of others who run into something similar, it was caused by a new "Workspace" key mapping:

```json
  {
    "context": "Workspace",
    "bindings": {
      "space g g": ["task::Spawn", { "gitu": "run gitu in current project" }]
    }
```

In hindsight it was obvious, but I was thrown off by the fact I had previously used "space-first" keybindings like that (I'm a recovering space/doom emacs junkie) but they'd all been in the `Editor && (vim_mode == normal || vim_mode == visual) && !VimWaiting && !menu` context, not the `workspace` context.

Anyway, I'll close this. Sorry about that.
